pshtt/pshtt.py

```diff
@@ -289,69 +314,97 @@ def basic_check(endpoint):
-        return
+    try:
+        if endpoint.https_bad_hostname:
+            endpoint.hsts = False
+            return
 
-    header = endpoint.headers.get("Strict-Transport-Security")
+        header = endpoint.headers.get("Strict-Transport-Security")
 
-    if header is None:
-        endpoint.hsts = False
-        return
+        if header is None:
+            endpoint.hsts = False
+            return
 
-    endpoint.hsts = True
-    endpoint.hsts_header = header
+        endpoint.hsts = True
+        endpoint.hsts_header = header
 
-    # Set max age to the string after max-age
-    # TODO: make this more resilient to pathological HSTS headers.
+        # Set max age to the string after max-age
+        # TODO: make this more resilient to pathological HSTS headers.
 
-    # handle multiple HSTS headers, requests comma-separates them
-    first_pass = re.split(',\s?', header)[0]
-    second_pass = re.sub('\'', '', first_pass)
+        # handle multiple HSTS headers, requests comma-separates them
+        first_pass = re.split(',\s?', header)[0]
+        second_pass = re.sub('\'', '', first_pass)
 
-    temp = re.split(';\s?', second_pass)
+        temp = re.split(';\s?', second_pass)
 
-    if "max-age" in header.lower():
-        endpoint.hsts_max_age = int(temp[0][len("max-age="):])
+        if "max-age" in header.lower():
+            endpoint.hsts_max_age = int(temp[0][len("max-age="):])
 
-    if endpoint.hsts_max_age <= 0:
-        endpoint.hsts = False
-        return
+        if endpoint.hsts_max_age <= 0:
+            endpoint.hsts = False
+            return
 
-    # check if hsts includes sub domains
-    if 'includesubdomains' in header.lower():
-        endpoint.hsts_all_subdomains = True
+        # check if hsts includes sub domains
+        if 'includesubdomains' in header.lower():
+            endpoint.hsts_all_subdomains = True
 
-    # Check is hsts has the preload flag
-    if 'preload' in header.lower():
-        endpoint.hsts_preload = True
+        # Check is hsts has the preload flag
+        if 'preload' in header.lower():
+            endpoint.hsts_preload = True
+    except Exception as err:
+        endpoint.unknownerror = True
+        logging.warn("Unknown exception when handling hsts check.")
+        logging.debug("{0}".format(err))
+        return
 
 
 # Uses sslyze to figure out the reason the endpoint wouldn't verify.
 def https_check(endpoint):
     logging.debug("sslyzing %s..." % endpoint.url)
 
     # remove the https:// from prefix for sslyze
-    hostname = endpoint.url[8:]
-    server_info = sslyze.server_connectivity.ServerConnectivityInfo(hostname=hostname, port=443)
+    try:
+        hostname = endpoint.url[8:]
+        server_info = sslyze.server_connectivity.ServerConnectivityInfo(hostname=hostname, port=443)
+    except Exception as err:
+        endpoint.unknownerror = True
+        logging.warn("Unknown exception when checking server connectivity info\
+            with sslyze.")
```

Comment:
Just noting that we don't impose a column limit in the tool or in our flake8 rules, so this can be one line.